packages/@glimmer/runtime/lib/vm/rehydrate-builder.ts

```diff
@@ -35,9 +36,7 @@ export class RehydratingCursor extends CursorImpl {
 export class RehydrateBuilder extends NewElementBuilder implements ElementBuilder {
   private unmatchedAttributes: Option<SimpleAttr[]> = null;
   [CURSOR_STACK]!: Stack<RehydratingCursor>; // Hides property on base class
-  private blockDepth = 0;
-
-  // private candidate: Option<SimpleNode> = null;
+  blockDepth = 0;
```

Comment:
Is the only reason why this has to be `public` is to enforce the type on L101?

Reply:
Yes, if it is `private` the "special" types that we added for `pushElement` won't work. The real issue is that `NewElementBuilder` (our super class) invokes `this.pushElement` during `RehydrateBuilder`'s `super()` call which means our class fields haven't ran yet (and therefore `this.blockDepth` is `undefined` still). So we tried to make the types such that it was clearer that the `this` within that method was "weird" (to avoid future mistakes).